src/Uno.UI/UI/Xaml/Controls/Image/ImageSizeHelper.cs

```diff
@@ -13,28 +14,53 @@ public static void MeasureSource(this Image image, Size finalSize, ref Rect chil
 			switch (image.Stretch)
 			{
 				case UniformToFill:
-					var uniformToFillScale = (child.Width * finalSize.Height >= child.Height * finalSize.Width)
-						? finalSize.Height / child.Height // child is flatter than parent
-						: finalSize.Width / child.Width; // child is taller than parent
-					child.Width *= uniformToFillScale;
-					child.Height *= uniformToFillScale;
-					break;
+					{
```

Comment:
These modifications don't impact existing unit tests for this class ?

Reply:
All tests are passing.

Reply:
Does it mean that there are missing tests, or were the modifications only refactoring ?

Reply:
Yes it's a refactoring. The output is supposed the be exactly the same.